content/knowledge-white-label/white-label-scripts.md

```diff
@@ -0,0 +1,216 @@
+
+# cURL request to download the file to the build machine
+curl -H "Authorization: Bearer $CONTENTFUL_API_TOKEN" $FILE_URL --output assets.zip
+{{< /highlight >}}
+
+
+## Changing app icons
+
+For iOS apps, ff you look at an Xcode project using Finder you will see that the icons added in Xcode are located in `<project-name>/<scheme-name>/Assets.xcassets/AppIcon.appiconset`. This means that after downloading icon assets for a specific client’s build, you can change them on disk by simply deleting the existing `AppIcon.appiconset` directory, and then copying the assets into the `Assets.xcassets` directory. 
+
+For example, you could do the following as one of your workflow steps:
+
+{{< highlight yaml "style=paraiso-dark">}}
+name: Change iOS app icons
+script: | 
+  # delete the existing icons
+  rm -rf ios/Runner/Assets.xcassets/AppIcon.appiconset
+  # copy the downloaded icons to Assets.xcassets directory
+  cp -r ./client_assets/AppIcon.appiconset ios/Runner/Assets.xcassets/
+{{< /highlight >}}
+
+For Android apps, you should run a similar script to update the icons located in app/src/main/res where you will find a number of directories that contain an icon for specific resolutions such as drawable-hdpi, drawable-mdpi, drawable-xhdpi, drawable-xxhdpi, drawable-xxxhdpi . Your script to update the icons in your Android project might look something like this:
+
+{{< highlight yaml "style=paraiso-dark">}}
+name: Change Android app icons
+script: |
+  unzip android_assets.zip -d android_assets
+  cp -r ./android_assets/* ./myapp/android/app/src/main/res
+{{< /highlight >}}
+
+## Automatic build versioning
+
+Each new app version that is published to Google Play or the Apple App Store needs to have a unique build number. You can use Codemagic’s CLI tools to retrieve the previous build number and then increment this for each new build. For example, the following shows how to increment the build number when building a Flutter iOS apps:
+
+{{< highlight yaml "style=paraiso-dark">}}
+name: Flutter build ipa and automatic versioning
+script: | 
+  flutter build ipa --release \
+    --build-name=1.0.0 \
+    --build-number=$(($(app-store-connect get-latest-testflight-build-number "$APP_STORE_ID") + 1)) \
+    --export-options-plist=/Users/builder/export_options.plist
+{{< /highlight >}}
+
+## Triggering builds with the Codemagic REST API
+
+The Codemagic REST API is used in a white label workflow to trigger builds for each unique client version you need to build. When triggering a build you can pass environment variables which identify a specific client, so their unique assets can be downloaded and used for the build. It can be as simple as passing the ID number associated with the client. 
```

Comment:
```suggestion
The Codemagic REST API is used in a white label workflow to trigger builds for each unique client version you need to build. When triggering a build, you can pass environment variables which identify a specific client so their unique assets can be downloaded and used for the build. It can be as simple as passing the ID number associated with the client. 
```